common/djangoapps/student/views/dashboard.py

```diff
@@ -225,6 +225,50 @@ def get_course_enrollments(user, org_whitelist, org_blacklist):
             yield enrollment
 
 
+def get_filtered_course_entitlements(user, org_whitelist, org_blacklist):
+    """
+    Given a user, return a filtered set of his or her course enrollments.
+
+    Arguments:
+
+    Returns:
+        generator[CourseEntitlement]: a sequence of enrollments to be displayed
+        on the user's dashboard.
+    """
+    course_entitlement_available_sessions = {}
+    unfulfilled_entitlement_pseudo_sessions = {}
+    course_entitlements = list(CourseEntitlement.get_active_entitlements_for_user(user))
+    filtered_entitlements = []
+    pseudo_session = None
+    course_key_str = ''
+
+    for course_entitlement in course_entitlements:
+        course_entitlement.update_expired_at()
+        available_runs = get_visible_sessions_for_entitlement(course_entitlement)
+
+        if not course_entitlement.enrollment_course_run or not available_runs:
+            # Unfulfilled entitlements need a mock session for metadata
+            pseudo_session = get_pseudo_session_for_entitlement(course_entitlement)
+            unfulfilled_entitlement_pseudo_sessions[str(course_entitlement.uuid)] = pseudo_session
+
+        # Check the org of the Course and filter out entitlements that are not available.
+        if available_runs:
+            course_key_str = available_runs[0]['key']
+        elif pseudo_session:
+            course_key_str = pseudo_session['key']
+
+        course_run_key = CourseKey.from_string(course_key_str)
+        if org_whitelist and course_run_key.org not in org_whitelist:
```

Comment:
When you get to this line it's possible that the course run key will still be an empty string. In the event that there are no available runs and `get_pseudo_session_for_entitlement` fails to retrieve a session.